src/components/PeopleTable/PeopleTable.tsx

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+
+import { PersonItem } from '../PersonItem';
+import { ColumnHeaders, Person } from '../../types';
+import { SearchLink } from '../SearchLink/SearchLink';
+import { useSearchParams } from 'react-router-dom';
+import classNames from 'classnames';
+
+interface Props {
+  people: Person[];
+}
+
+const DESCENDING_ORDER = 'desc';
+
+export const PeopleTable: React.FC<Props> = ({ people }) => {
+  const [searchParams] = useSearchParams();
+
+  const sort = searchParams.get('sort') || null;
+  const order = searchParams.get('order') || null;
+
+  const handleSortChange = (sortBy: string) => {
+    const isSelected = sortBy === sort;
+
+    if ((!sort && !order) || !isSelected) {
+      return { sort: sortBy, order: null };
+    }
+
+    if (sort && !order && isSelected) {
+      return { sort: sortBy, order: DESCENDING_ORDER };
+    }
+
+    if (sort && order) {
+      return { sort: null, order: null };
+    }
+
+    return { sort: null, order: null };
+  };
+
+  const getSortedPeople = () => {
+    let sortedPeople = [...people];
+
+    if (sort) {
+      const normalizedSort = sort.charAt(0).toUpperCase() + sort.slice(1);
+
+      switch (normalizedSort) {
+        case ColumnHeaders.name:
+          sortedPeople = sortedPeople.sort((person1, person2) =>
```

Comment:
DRY. You repeat this part in every `case` ` sortedPeople = sortedPeople.sort((person1, person2) =>`